### DISCUSSION TITLE
how GPUBuffer becomes mapping again?

### DISCUSSION THREAD

**Discussion by sensetimeXsy** [gpuweb/gpuweb#2182] [Q&A]
when i create a GPUBuffer with mappedAtCreation be true,i send an arrayBuffer to objetc which return by getMappedRange().After that i call unmap().Buf how i reassign another arrayBuffer to this GPUBuffer?this means than how this GPUBuffer becomes mapping again?

**Comment by Kangz** [comment]
After a buffer has been unmapped, if that buffer has one of the MAP usages, then you can asynchronously map them again using `GPUBuffer.MapAsync`. Note however that most buffers you have will not be mappable (since they would be restricted to only that usage and `COPY*`). Instead you can choose either to use a mappable staging buffer then do a buffer to buffer copy, or to use `GPUQueue.WriteBuffer` to put data in the buffer well-ordered with other queue operations.

**Comment by sensetimeXsy** [comment]
Thank you very much.I solved it by using GPUQueue.WriteBuffe as you said~
